servicemanager/servicemanager.go

```diff
@@ -301,27 +344,22 @@ func (sm *ServiceManager) doInstallService(newService ServiceInfo, imageURL stri
 		}
 	}
 
-	log.WithFields(log.Fields{"serviceID": newService.ServiceID}).Debug("Install service")
-
 	newService.ImagePath, err = ioutil.TempDir(sm.servicesDir, "")
 	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	serviceSize, err := sm.extractPackageByURL(newService.ImagePath, sm.downloadDir, imageURL, fileInfo)
+	packageSize, spacePackage, err := sm.extractPackageByURL(newService.ImagePath, sm.downloadDir, imageURL, fileInfo)
 	if err != nil {
 		return aoserrors.Wrap(err)
 	}
 
-	defer func() {
-		var size int64
+	newService.Size = uint64(packageSize)
 
-		if err != nil {
-			size = serviceSize
-			_ = os.RemoveAll(newService.ImagePath)
-		}
+	var spaceService *spaceallocator.Space
 
-		sm.deallocateSpace(size)
+	defer func() {
+		acceptReleaseSpace(newService, spaceService, spacePackage, err)
```

Comment:
Same comment as for layer manager commit.

Reply:
Done